Forge Networking Remastered Unity/Assets/Bearded Man Studios Inc/Scripts/Networking/Forge/Networking/CachedFacepunchP2PClient.cs

```diff
@@ -0,0 +1,228 @@
+﻿//
+// System.Net.Sockets.UdpClient.cs
+//
+// Author:
+//	Gonzalo Paniagua Javier <gonzalo@ximian.com>
+//	Sridhar Kulkarni (sridharkulkarni@gmail.com)
+//	Marek Safar (marek.safar@gmail.com)
+//
+// Modified by:
+//	Brent Farris (brent@beardedmangames.com)
+//
+// Copyright (C) Ximian, Inc. http://www.ximian.com
+// Copyright 2011 Xamarin Inc.
+//
+
+//
+// Permission is hereby granted, free of charge, to any person obtaining
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+//
+
+#if !UNITY_WEBGL
+#if !NetFX_CORE
+#if FACEPUNCH_STEAMWORKS
+
+using Steamworks;
+using System;
+using System.Collections.Generic;
+using System.Net;
+using System.Net.Sockets;
+
+namespace BeardedManStudios.Forge.Networking
+{
+    public class CachedFacepunchP2PClient : IDisposable
+	{
+		public const char HOST_PORT_CHARACTER_SEPARATOR = '+';
+		private bool disposed = false;
+		private bool active = false;
+        private SteamId steamEndPoint;
+
+#region Instantiation
+        public CachedFacepunchP2PClient()
+		{
+            //steamendpoint = self
+		}
```

Comment:
Indentation above is incorrect. If the commented out code is not needed then please also remove that too.